src/clusterfuzz/_internal/system/fast_http.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tools for fast HTTP operations."""
+import asyncio
+from concurrent import futures
+import contextlib
+import itertools
+import multiprocessing
+from typing import List
+from typing import Tuple
+
+import aiohttp
+
+from clusterfuzz._internal.system import environment
+
+_POOL_SIZE = multiprocessing.cpu_count()
+
+
+@contextlib.contextmanager
+def _pool(pool_size=_POOL_SIZE):
+  if (environment.get_value('PY_UNITTESTS') or
+      environment.platform() == 'WINDOWS'):
+    yield futures.ThreadPoolExecutor(pool_size)
+  else:
+    yield futures.ProcessPoolExecutor(pool_size)
+
+
+def download_urls(urls: List[str], filepaths: List[str]) -> str:
+  """Downloads multiple |urls| to |filepaths| in parallel and
+  asynchronously. Tolerates errors. Returns a list of whether each
+  download was successful."""
+  # TOOD(metzman): Move the multiprocessing code over from storage.
+  assert len(urls) == len(filepaths)
+  chunks = []
```

Comment:
nit: is `batches` a better term here? 

`chunk` is overloaded in this file (in `_async_download_file` where it has a different meaning).